### PR TITLE
Migrating Bor Module 

### PR DESCRIPTION
**Pull request by saikumar0x1967**
#543

### CODE REVIEW

File: x/bor/keeper/keeper.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+)
+
+// Keeper stores all related data
+type Keeper struct {
+	cdc codec.BinaryMarshaler
+	sk  stakingKeeper.Keeper
+	// The (unexposed) keys used to access the stores from the Context.
+	storeKey sdk.StoreKey
+	// codespace
+	//codespace sdk.CodespaceType
```
Comment (ssandeep) [inline x/bor/keeper/keeper.go:42]: let's remove this

File: x/bor/keeper/keeper.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+		limit = 20
+	}
+
+	// get paginated iterator
+	iterator := hmTypes.KVStorePrefixIteratorPaginated(store, SpanPrefixKey, uint(page), uint(limit))
+
+	// loop through validators to get valid validators
+	for ; iterator.Valid(); iterator.Next() {
+		var span hmTypes.Span
+		if err := k.cdc.UnmarshalBinaryBare(iterator.Value(), &span); err == nil {
```
Comment (ssandeep) [inline x/bor/keeper/keeper.go:188]: should we return an error if `err != nil`

File: x/bor/keeper/keeper.go [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+	// loop through spans to get valid spans
+	for ; iterator.Valid(); iterator.Next() {
+		// unmarshall span
+		var result hmTypes.Span
+		if err := k.cdc.UnmarshalBinaryBare(iterator.Value(), &result); err != nil {
+			k.Logger(ctx).Error("Error UnmarshalBinaryBare", "error", err)
+		}
+		// call function and return if required
+		if err := f(result); err != nil {
+			return
```
Comment (ssandeep) [inline x/bor/keeper/keeper.go:350]: return err?

File: x/bor/side_handler.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	return func(ctx sdk.Context, msg sdk.Msg) abci.ResponseDeliverSideTx {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+		switch msg := msg.(type) {
+		case *types.MsgProposeSpan:
+			return SideHandleMsgSpan(ctx, k, *msg, contractCaller)
+		default:
+			errMsg := fmt.Sprintf("unrecognized %s message type: %T", types.ModuleName, msg)
+			fmt.Println(errMsg)
+			return abci.ResponseDeliverSideTx{
+				Code: uint32(6), // TODO should be changed like `sdk.CodeUnknownRequest `
```
Comment (ssandeep) [inline x/bor/side_handler.go:35]: Use `Code: sdkerrors.ErrUnknownRequest.ABCICode(),` instead.
Can you also change this in staking, clerk and checkpoint?

**Review by ssandeep — CHANGES_REQUESTED** [review]

File: x/bor/keeper/keeper.go (hunk 2 above)
Comment (ssandeep) [reply x/bor/keeper/keeper.go:188]: in the else clause, I mean.

**Comment by saikumar0x1967** [comment]
@ssandeep  review changes are done, please check once

**Review by ssandeep — APPROVED** [review]

File: x/bor/keeper/keeper.go (hunk 3 above)
Comment (ssandeep) [reply x/bor/keeper/keeper.go:353]: we need to return err here right @saikumar0x1967

File: x/bor/side_handler.go (hunk 4 above)
Comment (ssandeep) [reply x/bor/side_handler.go]: ok, other modules we'll take it up later.

**Review by ssandeep — APPROVED** [review]

**Review by ssandeep — APPROVED** [review]

**Pull request merged**